core/src/types/operator/operator.rs

```diff
@@ -544,6 +544,13 @@ impl Operator {
         self.write_with(path, OpWrite::new(), bs).await
     }
 
+    /// Copy a file from `from` to `to`.
+    pub async fn copy(&self, from: &str, to: &str) -> Result<()> {
```

Comment:
We do not support copying a directory in `copy`. Therefore, we need to ensure that the input path is not a directory.

Reply:
This is exactly what I was about to achieve. However, I can't actually call it, so I haven't dealt with the rest of the logic yet.

Reply:
Which part blocks you? I think we can copy code like: 

https://github.com/apache/incubator-opendal/blob/d5e0ccaae6feb6ef5278c40eb75c4bb3256bec73/core/src/types/operator/operator.rs#L641-L648

Reply:
Could you try it? I tried and had been able to only call the default implementation(with the `unsupported` error).
I don't know what I missed.

Reply:
> I think we can copy code like:

Yes. This is what it needs.

Reply:
> Could you try it? I tried and had been able to only call the default implementation(with the `unsupported` error). I don't know what I missed.

Ok, I got it. You need to add `copy` in [LayeredAccessor](https://opendal.apache.org/docs/rust/opendal/raw/trait.LayeredAccessor.html) and forword to inner.

Reply:
You are right. I missed the second part `impl<L: LayeredAccessor> Accessor for L`. It works.

I'll finish it tomorrow.